src/core/core.h

```
@@ -396,7 +401,7 @@ class System {
     struct Impl;
     std::unique_ptr<Impl> impl;
 
-    static System s_instance;
+    inline static std::unique_ptr<System> s_instance{};
```

This doesn't seem like a related change to the logging system.

---

If this change is undone yuzu will crash on launch because I'm now enforcing that we don't lose log messages on startup. Just delete the `Common::Log::InitializeLogging()` in `main.cpp` and run with gdb to see where we're losing messages.

System's constructor ends up indirectly logging some messages. Before this PR the logging silently fails. Now I delay and make the System initialization explicit instead of surprisingly happening before `main`. This rescues the early log messages.

---

I think its fine if we lose messages on startup for the time being, because we are going to change how the system instance is constructed after #6774 is merged. I'd suggest to keep the changes small in scope to avoid potential weird behavior